kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/api/AdapterGenerator.kt

```diff
@@ -211,12 +218,31 @@ internal class AdapterGenerator(
     // constructor arguments, while property index is only based on the index passed into
     // JsonReader.Options.
     var propertyIndex = 0
-    var maskIndex = 0
     val constructorPropertyTypes = mutableListOf<CodeBlock>()
+
+    //
+    // Track important indices for masks. Masks generally increment with each parameter (including
+    // transient).
+    //
+    // Mask name index is an index into the maskNames array we initialized above.
+    //
+    // Once the maskIndex reaches 32, we've filled up that mask and have to move to the next mask
+    // name. Reset the maskIndex relative to here and continue incrementing.
+    //
+    var maskIndex = 0
```

Comment:
Don't you want just one variable, maskableIndex? Then you can do (maskableIndex / 32) and (maskableIndex % 32) to get the bits you want.

Reply:
Tracking it this way feels awkward because it's less random-access to go from maskableIndex to the specific bits you want to adjust.

Reply:
Either way works, I just thought this was a bit more readable. Which do you rather?

Reply:
I think it’s less readable, but that’s cause I do lots of code like this and using divide+mod at read time feels natural to me.